QualityControl/public/services/Layout.service.js

```diff
@@ -64,19 +73,25 @@ export default class LayoutService {
   /**
    * Method to get all layouts by the user's id
    * @param {string} userId - user id for which to query layouts
+   * @param {string|undefined} fields - comma seperated string values. Represent the fields that should be fetched.
+   * If left empty all available fields will be fetched
    * @param {Class<Observable>} that - Observer requesting data that should be notified of changes
    * @returns {undefined}
    */
-  async getLayoutsByUserId(userId, that = this.model) {
+  async getLayoutsByUserId(userId, fields = undefined, that = this.model) {
     this.userList = RemoteData.loading();
     that.notify();
 
     if (isNaN(userId)) {
       this.userList = RemoteData.failure('Provided userId is not a number');
     } else {
-      const { result, ok } = await this.loader.get(`/api/layouts?owner_id=${userId}`);
+      const url = `/api/layouts?owner_id=${userId}${fields !== undefined ? `&fields=${fields}` : ''}`;
+
+      const { result, ok } = await this.loader.get(url);
       if (ok) {
-        const sortedLayouts = result.sort((lOne, lTwo) => lOne.name > lTwo.name ? 1 : -1);
+        const layoutComparitor = (layout1, layout2) => layout1.name.localeCompare(layout2.name);
```

Comment:
I see that you are implementing two comparitors which aim to sort by name but in two different manners:
Here and a few lines above. 

Best is to extract this in a util function and reuse it from there